From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Sat, 11 Jun 2016 21:37:29 +0200
Subject: [PATCH] Nailgun: Increase target image build timeout

Currently we're close to 40 minutes of image building (thanks to
qemu-debootstrap). If network connectivity is a bit slow (happened to
me) it's easy to go over 1h limit. Therefore let's push this to 2 hours
to be safer.

[ Alexandru Avadanii ]
Moved original patch wrote by Stan to post-install.

Signed-off-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 .../f_isoroot/f_bootstrap/post-scripts/80_prepare_cross_builds.sh  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/build/f_isoroot/f_bootstrap/post-scripts/80_prepare_cross_builds.sh b/build/f_isoroot/f_bootstrap/post-scripts/80_prepare_cross_builds.sh
index 3cb22f3..c106bc3 100755
--- a/build/f_isoroot/f_bootstrap/post-scripts/80_prepare_cross_builds.sh
+++ b/build/f_isoroot/f_bootstrap/post-scripts/80_prepare_cross_builds.sh
@@ -45,6 +45,13 @@ if which dpkg > /dev/null 2>&1; then
     # the maximum timeout value for mcagent has to be increased (use 2h).
     sed -i.bak -r 's/^(\s+:timeout\s*=>)\s*[[:digit:]]+$/\1 7200/' \
       /usr/libexec/mcollective/mcollective/agent/execute_shell_command.ddl
+    # Update nailgun timeout setting (2h)
+    if [ -f ${PYTHON_SITEPKGS_DIR}/nailgun/settings.yaml ]; then
+      sed -i.bak -r \
+        's/^(PROVISIONING_IMAGES_BUILD_TIMEOUT:)\s*[[:digit:]]+$/\1 7200/' \
+        ${PYTHON_SITEPKGS_DIR}/nailgun/settings.yaml
+      echo "INFO: [1h] => [2h] updated nailgun image build timeout."
+    fi

     # Bootstrap: Use gzip instead of xz compression.
     #
